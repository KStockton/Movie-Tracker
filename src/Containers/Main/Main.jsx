import React, { Component } from "react";
import Home from "../Home/Home";
import { connect } from "react-redux";


class Main extends Component {
  render() {
    return (
      <main>
        <Home />
      </main>
    );
  }
}

const mapStateToProps = state => ({
  user: state.user
});



export default connect(mapStateToProps)(Main)
