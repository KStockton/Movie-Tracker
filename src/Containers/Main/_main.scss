

.no-favorites{
    height: 100vh;
}