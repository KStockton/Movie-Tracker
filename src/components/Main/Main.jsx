import React, { Component } from "react";
import Nav from "../Nav/Nav";
import Footer from "../Footer/Footer";
import Home from "../Home/Home";
import SearchResults from "../SearchResults/SearchResults";
import { connect } from "react-redux";
import UserSignIn from "../UserSignIn/UserSignIn";
import UserSettings from "../UserSettings/UserSettings";
import { Route, Redirect } from "react-router-dom";


class Main extends Component {
  constructor() {
    super();
    this.state = {};
  }

  async componentDidMount() {}


  handleFilterClick = criteria => {
    this.setState({ criteria });
  };
  render() {
    console.log(this.props.user.name);
    console.log("rendering");
    return (
      <main>
        <Nav
          handleFilterClick={this.handleFilterClick}
          handleUserClick={this.handleUserClick}
        />
        <Home />
        <Footer />
      </main>
    );
  }
}

const mapStateToProps = state => ({
  user: state.user
});

const mapDispatchToProps = dispatch => ({});


export default connect(mapStateToProps)(Main)
