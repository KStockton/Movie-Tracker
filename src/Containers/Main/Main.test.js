import React from "react";
import { Main, mapStateToProps } from "./Main";
import { shallow } from "enzyme";

describe("Main", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Main />);
  });

  it("Main component should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should return a props object ", () => {
    const mockState = { user: { name: "max", id: 1 }, removeUser: jest.fn() };
    const expected = {
      user: mockState.user
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});
