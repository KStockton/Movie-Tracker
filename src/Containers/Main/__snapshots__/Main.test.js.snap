// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Main component should match the snapshot 1`] = `
<main>
  <ConnectFunction />
</main>
`;
